src/c++/perf_analyzer/genai-perf/genai_perf/main.py

```diff
@@ -54,21 +54,25 @@ def create_artifacts_dirs(generate_plots: bool) -> None:
 
 
 def generate_inputs(args: Namespace, tokenizer: Tokenizer) -> None:
-    # TODO (TMA-1758): remove once file support is implemented
-    input_file_name = ""
     # TODO (TMA-1759): review if add_model_name is always true
     add_model_name = True
     try:
-        extra_input_dict = parser.get_extra_inputs_as_dict(args)
+        extra_input_dict = (
+            parser.get_extra_inputs_as_dict(args)
+            if args.extra_inputs is not None
+            else None
+        )
     except ValueError as e:
         raise GenAIPerfException(e)
 
     LlmInputs.create_llm_inputs(
         input_type=args.prompt_source,
         output_format=args.output_format,
-        dataset_name=args.input_dataset,
+        dataset_name=args.input_dataset or "",
```

Comment:
If this is defined, it will be set and the infer logic will set the prompt source to dataset based on how it is set up.
Would you prefer the infer logic is more through and this becomes a pass thru?

Reply:
I see. I think I'd prefer the underlying logic gets looked at. I don't understand why the default value for this arg isn't None (or an empty string, if needed), and that gets passed rather than extra logic happening at the args-being-passed-into-a-function-call level.

Reply:
Great work!